File: backend/src/main/java/com/project/dto/professional/request/ProfessionalUpdateDTO.java
package com.project.dto.professional.request;

import java.math.BigDecimal;

import com.project.enums.Gender;
import com.project.enums.Specialization;
import com.project.enums.SpokenLanguage;

import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;

import lombok.Data;

@Data
public class ProfessionalUpdateDTO {

    // USER fields (optional but validated if present)

    @Size(min = 3, max = 50, message = "Full name must be between 3 and 50 characters")
    @Pattern(
        regexp = "^[A-Za-z ]+$",
        message = "Full name can contain only letters and spaces"
    )
    private String fullName;

    @Email(message = "Invalid email format")
    private String email;

    @Pattern(
        regexp = "^[6-9]\\d{9}$",
        message = "Phone number must be a valid 10-digit Indian mobile number"
    )
    private String phone;

    private Gender gender;

    // PROFESSIONAL fields

    private Specialization specialization;

    private SpokenLanguage spokenLanguage;

    @Min(value = 0, message = "Experience cannot be negative")
    @Max(value = 60, message = "Experience seems invalid")
    private Integer experienceYears;

    @Size(min = 2, max = 100, message = "Qualification must be between 2 and 100 characters")
    private String qualification;

    @Size(max = 500, message = "Bio must not exceed 500 characters")
    private String bio;

    @DecimalMin(value = "0.0", inclusive = false, message = "Consultation fee must be greater than 0")
    private BigDecimal consultationFee;
}

File: backend_dotnet/SafeMind.Core/DTOs/Professional/Request/ProfessionalUpdateDTO.cs
using System.ComponentModel.DataAnnotations;
using SafeMind.Core.Enums;

namespace SafeMind.Core.DTOs.Professional.Request
{
    public class ProfessionalUpdateDTO
    {
        [StringLength(50, MinimumLength = 3, ErrorMessage = "Full name must be between 3 and 50 characters")]
        [RegularExpression(@"^[A-Za-z ]+$", ErrorMessage = "Full name can contain only letters and spaces")]
        public string? FullName { get; set; }

        [EmailAddress(ErrorMessage = "Invalid email format")]
        public string? Email { get; set; }

        [RegularExpression(@"^[6-9]\d{9}$", ErrorMessage = "Phone number must be a valid 10-digit Indian mobile number")]
        public string? Phone { get; set; }

        public Gender? Gender { get; set; }

        public Specialization? Specialization { get; set; }
        public SpokenLanguage? SpokenLanguage { get; set; }

        [Range(0, 60)]
        public int? ExperienceYears { get; set; }

        [StringLength(100, MinimumLength = 2)]
        public string? Qualification { get; set; }

        [StringLength(500)]
        public string? Bio { get; set; }

        [Range(0.1, double.MaxValue)]
        public decimal? ConsultationFee { get; set; }
    }
}
